modules/wss4j/src/org/apache/ws/security/saml/SAMLIssuerImpl.java

```diff
@@ -148,110 +146,94 @@ public SAMLAssertion newAssertion() { // throws Exception {
-                            null,
-                            Arrays.asList(statements));
-
-            if (!senderVouches) {
-                KeyInfo ki = new KeyInfo(instanceDoc);
-                try {
-                    X509Certificate[] certs =
-                            userCrypto.getCertificates(username);
-                    if (sendKeyValue) {
-                        PublicKey key = certs[0].getPublicKey();
-                        String pubKeyAlgo = key.getAlgorithm();
-                        
-                        if ("DSA".equalsIgnoreCase(pubKeyAlgo)) {
-                            DSAKeyValue dsaKeyValue = new DSAKeyValue(instanceDoc, key);
-                            ki.add(dsaKeyValue);
-                        } else if ("RSA".equalsIgnoreCase(pubKeyAlgo)) {
-                            RSAKeyValue rsaKeyValue = new RSAKeyValue(instanceDoc, key);
-                            ki.add(rsaKeyValue);
-                        }
-                    } else {
-                        X509Data certElem = new X509Data(instanceDoc);
-                        certElem.addCertificate(certs[0]);
-                        ki.add(certElem);
-                    }
-                } catch (WSSecurityException ex) {
-                    if (log.isDebugEnabled()) {
-                        log.debug(ex.getMessage(), ex);
-                    }
-                    return null;
-                } catch (XMLSecurityException ex) {
-                    if (log.isDebugEnabled()) {
-                        log.debug(ex.getMessage(), ex);
+        NameIdentifier nameId = SAMLUtil.newSamlObject(NameIdentifier.class);
+        nameId.setValue(name);
+        nameId.setNameQualifier(qualifier);
+
+        String subjectIP = null;
+        String authMethod = null;
+        if ("password"
+                .equals(properties.getProperty("org.apache.ws.security.saml.authenticationMethod"))) {
+            authMethod =
+                    AuthenticationStatement.PASSWORD_AUTHN_METHOD;
+        }
+        DateTime authInstant = new DateTime();
+
+        Subject subject = SAMLUtil.newSamlObject(Subject.class);
+        subject.setNameIdentifier(nameId);
+
+        AuthenticationStatement authenticationStatement = SAMLUtil.newSamlObject(AuthenticationStatement.class);
+        authenticationStatement.setAuthenticationMethod(authMethod);
+        authenticationStatement.setAuthenticationInstant(authInstant);
+
+
+        sa = SAMLUtil.newSamlObject(Assertion.class);
+        sa.setIssuer(issuer);
+
+        if (!senderVouches) {
+            KeyInfo ki = new KeyInfo(instanceDoc);
+            try {
+                X509Certificate[] certs =
+                        userCrypto.getCertificates(username);
+                if (sendKeyValue) {
+                    PublicKey key = certs[0].getPublicKey();
+                    String pubKeyAlgo = key.getAlgorithm();
+
+                    if ("DSA".equalsIgnoreCase(pubKeyAlgo)) {
```

Comment:
Use constants

Reply:
fixed in 2f696f6746f53ec59ed149ea7753dbcb09f7b044

Reply:
fixed in 2f696f6746f53ec59ed149ea7753dbcb09f7b044